HotelBooking/HotelBooking.BusinessLogic/Services/Implementation/ReservationService.cs

```diff
@@ -37,4 +47,49 @@ public async Task<ReservationDto> GetReservation(int id)
 
         return dto;
     }
+
+    public async Task AddReservation(CreateReservationDto createReservationDto)
+    {
+        await _validationService.Validate(createReservationDto);
+
+        var roomInfos = createReservationDto.Rooms;
+
+        var rooms = new List<Room>();
+        decimal price = 0;
+
+        foreach (var roomInfo in roomInfos)
+        {
+            var room = await _roomRepository.GetRoom((DataAccessLayer.Entities.RoomType)roomInfo.Type, roomInfo.Capacity, createReservationDto.DateFrom, createReservationDto.DateTo);
+            
+            if(room is null)
+            {
+                throw new NotFoundException("Selected rooms are not available on the specified date.");
+            }
+            else
+            {
+                price += room.PricePerNight * (createReservationDto.DateTo - createReservationDto.DateFrom).Days;
```

Comment:
Tutaj ze względu na to, że bierzemy też godziny, moge zrobić rezerwację na 1 minutę i cena wtedy wynosi 0. To kwestia tego, że nie powinniśmy brać pod uwagę godziny, to co gdzieś tam wyżej pisałem. Może typ DateOnly, albo coś pokombinować innego